src/js/providers/html5.js

```diff
@@ -476,9 +476,16 @@ function VideoProvider(_playerId, _playerConfig, mediaElement) {
                     currentQuality = i;
                 }
                 if (label && levels[i].label === label) {
-                    return i;
+                    currentQuality = i;
+                    break;
                 }
             }
+            if (!currentQuality) {
+                // Choose the highest quality
+                const sorted = levels.map((level, index) => Object.assign({ index, height: 0 }, level))
+                    .sort((a, b) => b.height - a.height);
+                currentQuality = sorted[0].index;
+            }
```

Comment:
This block was added for demo purposes on the 2016 device. 2018-19 devices support adaptive hls playback with the hls.js provider (0.12.4 fork).